test/cypress/integration/simplified-account/service-settings/api-keys/api-keys.cy.js

```diff
@@ -169,6 +169,49 @@ describe('Settings - API keys', () => {
       })
     })
 
+    describe('revoke an api key', () => {
+      const TOKEN_LINK = 'token-link-2'
+      const DESCRIPTION = 'my api key'
+      const apiKeys = [
+        new Token().withCreatedBy('joe bloggs').withDescription(DESCRIPTION)
+          .withIssuedDate('10 Dec 2024').withLastUsed('10 Dec 2024').withTokenLink(TOKEN_LINK)
+      ]
+
+      beforeEach(() => {
+        setupStubs('admin', apiKeys)
+        cy.task('setupStubs', [
+          apiKeysStubs.getKeyByTokenLink(GATEWAY_ACCOUNT_ID, TOKEN_LINK, DESCRIPTION),
+          apiKeysStubs.revokeKey(GATEWAY_ACCOUNT_ID, TOKEN_LINK)
+        ])
+      })
+
+      it('should revoke the api key successfully', () => {
+        cy.visit(`/simplified/service/${SERVICE_EXTERNAL_ID}/account/${ACCOUNT_TYPE}/settings/api-keys`)
+        cy.get('.govuk-summary-card').within(() => {
+          cy.contains('h2', DESCRIPTION).should('exist')
+          cy.contains('a', 'Revoke').click()
+        })
+        cy.get('input[type="radio"][value="Yes"]').check()
+        cy.contains('button', 'Save changes').click()
+        cy.url().should('include', `/simplified/service/${SERVICE_EXTERNAL_ID}/account/${ACCOUNT_TYPE}/settings/api-keys`)
+        cy.contains('h1', 'Test API keys').should('exist')
+        cy.contains('p.govuk-notification-banner__heading', `${DESCRIPTION} was successfully revoked`).should('exist')
+      })
+
+      it('should not revoke the api key successfully', () => {
+        cy.visit(`/simplified/service/${SERVICE_EXTERNAL_ID}/account/${ACCOUNT_TYPE}/settings/api-keys`)
+        cy.get('.govuk-summary-card').within(() => {
+          cy.contains('h2', DESCRIPTION).should('exist')
+          cy.contains('a', 'Revoke').click()
+        })
+        cy.get('input[type="radio"][value="No"]').check()
+        cy.contains('button', 'Save changes').click()
+        cy.url().should('include', `/simplified/service/${SERVICE_EXTERNAL_ID}/account/${ACCOUNT_TYPE}/settings/api-keys`)
+        cy.contains('h1', 'Test API keys').should('exist')
+        cy.contains('p.govuk-notification-banner__heading', `${DESCRIPTION} was successfully revoked`).should('not.exist')
+      })
+    })
+
```

Comment:
Is it worth adding a test for if nothing is selected as well?

Reply:
Done